### ISSUE TITLE
Issue when adding toctree with :hidde:

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1263]
I was making some tests and added the following in my index.rst file:

```
.. toctree::
   :hidden:

   references
```

When builidng the documentation using `make html` I got the following error:

```
#!

Exception occurred:
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/environment.py", line 1046, in get_toctree_for
    result.extend(toctree.children)
AttributeError: 'NoneType' object has no attribute 'children'
The full traceback has been saved in /tmp/sphinx-err-D_zjSJ.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
Either send bugs to the mailing list at <http://groups.google.com/group/sphinx-users/>,
or report them in the tracker at <https://bitbucket.org/birkenfeld/sphinx/issues/>. Thanks!
make: *** [html] Error 1

```

The content of `/tmp/sphinx-err-D_zjSJ.log` is the following:

```
# Sphinx version: 1.2b1
# Python version: 2.7.3
# Docutils version: 0.10 release
# Jinja2 version: 2.6
Traceback (most recent call last):
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/cmdline.py", line 247, in main
    app.build(force_all, filenames)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/application.py", line 211, in build
    self.builder.build_update()
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/builders/__init__.py", line 211, in build_update
    'out of date' &#37; len(to_build))
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/builders/__init__.py", line 267, in build
    self.write(docnames, list(updated_docnames), method)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/builders/__init__.py", line 310, in write
    self._write_serial(sorted(docnames), warnings)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/builders/__init__.py", line 323, in _write_serial
    self.write_doc(docname, doctree)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/builders/html.py", line 438, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/builders/html.py", line 750, in handle_page
    output = self.templates.render(templatename, ctx)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/jinja2glue.py", line 128, in render
    return self.environment.get_template(template).render(context)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Jinja2-2.6-py2.7.egg/jinja2/environment.py", line 894, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/themes/basic/page.html", line 10, in top-level template code
    {&#37;- extends "layout.html" &#37;}
  File "/home/lcerone/CNOVE/lib/python2.7/site-packages/easydev-0.5.9-py2.7.egg/share/themes/cno/layout.html", line 1, in top-level template code
    {&#37; extends "agogo/layout.html" &#37;}
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/themes/agogo/layout.html", line 11, in top-level template code
    {&#37;- extends "basic/layout.html" &#37;}
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/themes/basic/layout.html", line 160, in top-level template code
    {&#37;- block content &#37;}
  File "/home/lcerone/CNOVE/lib/python2.7/site-packages/easydev-0.5.9-py2.7.egg/share/themes/cno/layout.html", line 105, in block "content"
    {&#37;- block sidebarsearch &#37;}
  File "/home/lcerone/CNOVE/lib/python2.7/site-packages/easydev-0.5.9-py2.7.egg/share/themes/cno/layout.html", line 115, in block "sidebarsearch"
    {&#37;- block sidebartoc &#37;}
  File "/home/lcerone/CNOVE/lib/python2.7/site-packages/easydev-0.5.9-py2.7.egg/share/themes/cno/layout.html", line 117, in block "sidebartoc"
    {{ toctree() }}
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Jinja2-2.6-py2.7.egg/jinja2/sandbox.py", line 349, in call
    return __context.call(__obj, *args, **kwargs)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/builders/html.py", line 742, in <lambda>
    ctx['toctree'] = lambda **kw: self._get_local_toctree(pagename, **kw)
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/builders/html.py", line 678, in _get_local_toctree
    docname, self, collapse, **kwds))['fragment']
  File "/home/lcerone/CNOVE/local/lib/python2.7/site-packages/Sphinx-1.2b1-py2.7.egg/sphinx/environment.py", line 1046, in get_toctree_for
    result.extend(toctree.children)
AttributeError: 'NoneType' object has no attribute 'children'
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1263
- Originally reported by: Luca Cerone
- Originally created at: 2013-09-17T15:21:53.868

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-09-17 13:45:14+00:00_

Duplicate of #1157.

**Comment by shimizukawa** [comment]
_From Luca Cerone on 2013-09-17 14:06:52+00:00_

Hi Takayuki,
thanks for the review.

I am sorry but I don't understand:
I am experiencing this in a version 1.2b1, while you marked it as a duplicate of a bug that was solved in v.1.1.3. 

Is this really the same issue? if so shouldn't it be re-opened? if not, I think it might be a different bug.

Thanks for your help!

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-09-17 14:27:33+00:00_

Hi [Luca Cerone](https://bitbucket.org/lucacerone)

```
> I am sorry but I don't understand: I am experiencing this in a version 1.2b1,
> while you marked it as a duplicate of a bug that was solved in v.1.1.3.

> Is this really the same issue? 
```

I think so.

1.2b1 released at 82674d95 (2013-03-31), and my fix committed at 48a41f195c27 (2013-04-30),

Just now 1.2b2 released, please check it!
https://pypi.python.org/pypi/Sphinx/1.2b2

**Comment by shimizukawa** [comment]
_From Luca Cerone on 2013-09-17 14:43:03+00:00_

Hi Takayaki,
thanks for the explanation!

I upgraded to 1.2b2 and now the build proceeds flawlessly!
Thanks a lot again for the fix and for the help!
